marklogic-data-hub-central/ui/src/components/entities/mapping/entity-map-table/entity-map-table.tsx

```diff
@@ -602,35 +599,34 @@ const EntityMapTable: React.FC<Props> = (props) => {
     </div>
   );
 
-  const handleOptionSelect = (selectedValues) => {
+  const handleOptionSelect = (newlySelectedValues) => {
     let selectedArray: any = [];
-    let entityArray = props.isRelatedEntity ? props.includedEntityTypeProperties : props.relatedEntityTypeProperties;
+    let entityArray = props.relatedEntityTypeProperties;
 
-    if (selectedValues.length !== 0) {
+    if (newlySelectedValues.length !== 0) {
       //in the properties array, push the object that has the key which matches the value of the entity name selected
-      selectedValues.forEach(val => {
+      newlySelectedValues.forEach(val => {
         let index = entityArray.findIndex(object => object.entityLabel === val);
-        selectedArray.push(entityArray[index]);
+        if (!props.relatedEntitiesSelected.includes((entityArray[index]))) {
+          selectedArray.push(entityArray[index]);
+        }
       });
+      props.setRelatedEntitiesSelected(prevState => ([...prevState, ...selectedArray]));
     } else {
```

Comment:
Thanks @wooldridge. Regarding that line, it is actually needed to update the parent so it can keep track of all related entities that are currently selected. The selectedArray is only limited to maintaining which options are selected respective to each individual table.

Reply:
Misunderstood the original comment because the line above it was highlighted. Removed the else statement as it was unnecessary! Thanks @wooldridge .